### PR TITLE
Return success from set

### PR DESCRIPTION
**Pull request by skalsi-atlassian**
Hi Pamela,

This PR is to return a boolean from the `set` method with whether the data was inserted correctly. This can be handy sometimes for metrics or for external optimisations, chaining caches, etc.

Please let me know if there are any changes I can make to get this merged.

Thanks,
Sunny

### CODE REVIEW

**Comment by pamelafox** [comment]
Sorry, I didn't notice this PR. This seems fairly backwards-incompatibly friendly, and is how memcache does it. Hm. Are there any drawbacks to this that you can think of? (If you're still around, sorry!)

**Comment by skalsi-atlassian** [comment]
Hello. I think this is a pretty straight extension of the API, so it should be fine from my understanding. I'd love to get this merged :)

File: tests/tests.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ var startTests = function (lscache) {
   test('Testing set() and get() with string', function() {
     var key = 'thekey';
     var value = 'thevalue';
-    lscache.set(key, value, 1);
-    if (lscache.supported()) {
+    var isset = lscache.set(key, value, 1);
```
Comment (pamelafox) [inline tests/tests.js]: *nit* can you camelCase isSet?

File: tests/tests.js [hunk 2]
```diff
@@ -180,7 +180,7 @@ var startTests = function (lscache) {
       var key = 'thekey';
       var value = 'thevalue';
       var minutes = 1;
-      lscache.set(key, value, minutes);
+      equal(lscache.set(key, value, minutes), true, 'We expect the value to be inserted successfully');
       setTimeout(function() {
         equal(lscache.get(key), null, 'We expect value to be null');
         start();
```
Comment (pamelafox) [inline tests/tests.js]: There don't seem to be any tests that check for when it returns false, is it possible to add them, or is it too hard/not useful to mock the necessary environment for failures to happen?

Comment (skalsi-atlassian) [reply tests/tests.js]: I thought I'd have time to address these today but unfortunately not. Please hang out for a bit longer, I'll try and get it done in the next week or so. Thanks.

Comment (skalsi-atlassian) [reply tests/tests.js]: OK maybe that was a little bit Valve time but I got it done in the end. I've tested the scenarios which can reasonably be tested. I don't think we can test the other ones (disabled features, generic exceptions), unless you can think of a way.

**Comment by skalsi-atlassian** [comment]
Sorry for the wait. I've updated the PR with review comments.

File: tests/tests.js [hunk 3]
```diff
@@ -192,13 +202,36 @@ var startTests = function (lscache) {
...
+      while(num < 10000) {
+        try {
+          localStorage.setItem(key + num, longString);
+          num++;
+        } catch (e) {
+          break;
+        }
+      }
+      localStorage.clear();
+      // Now make string long enough to go over limit.
```
[the remark this answers is not in the document]
Comment (pamelafox) [reply tests/tests.js]: @skalsi-atlassian I just merged this and the tests aren't passing for me:
>> lscache - Testing set() and get() with string and expiration
>> Message: Expected 1 assertions, but 2 were run

Does that one pass for you?

[the remark this answers is not in the document]
Comment (pamelafox) [reply tests/tests.js]: (And yes, I should hook up TravisCi to run the tests for me. One day)

[the remark this answers is not in the document]
Comment (skalsi-atlassian) [reply tests/tests.js]: Sorry for the late reply. I just ran the tests in Firefox manually, and they were passing for me IIRC. I _think_ I had an issue with one of the timing based tests but I don't think it was one I touched.

**Pull request merged**